docs/dg/dev/architecture/api-platform/troubleshooting.md

```diff
@@ -0,0 +1,471 @@
+
+**Possible causes:**
+
+1. Validation schema file not found
+2. Wrong operation name in validation schema
+3. Validation groups not matching
+
+**Solution:**
+
+1. Ensure validation file exists:
+
+   ```bash
+   ✅ resources/api/backoffice/customers.validation.yml
+   ```
+
+2. Match operation names to HTTP methods:
+
+   ```yaml
+   post:      # For POST /customers
+     email:
+       - NotBlank
+
+   patch:     # For PATCH /customers/{id}
+     email:
+       - Optional:
+           constraints:
+             - Email
+   ```
+
+3. Check generated resource class has validation attributes:
+
+   ```php
+   #[Assert\NotBlank(groups: ['customers:create'])]
+   #[Assert\Email(groups: ['customers:create'])]
+   public ?string $email = null;
+   ```
+
+### API documentation UI not displaying correctly
+
+**Symptom:** When accessing the root URL of your API application, you see:
+- Missing styles/CSS
+- Broken JavaScript functionality
+- Plain HTML without formatting
+- "Failed to load resource" errors in browser docker/sdk cli glue 
+
+**Cause:** Assets were not installed after API Platform integration.
+
+**Solution:**
+
+Run the appropriate assets:install command for your application:
+
+### For Glue application
+
+```bash
+docker/sdk cli glue assets:install public/Glue/assets  --symlink
+```
+
+### For GlueStorefront
+
+```bash
+docker/sdk cli GLUE_APPLICATION=GLUE_STOREFRONT glue assets:install public/GlueStorefront/assets/  --symlink
+```
+
+### For GlueBackend
+
+```bash
+docker/sdk cli GLUE_APPLICATION=GLUE_BACKEND glue assets:install public/GlueBackend/assets/  --symlink
+```
+
+Then verify the documentation UI loads correctly by visiting the root URL:
+- Storefront: `https://glue.mysprykershop.com/`
+- Backoffice: `https://backoffice.mysprykershop.com/`
+
+{% info_block warningBox "Required after integration" %}
+
+The `assets:install` command must be run after integrating API Platform and whenever API Platform assets are updated. This is a required step documented in [How to integrate API Platform](/docs/dg/dev/upgrade-and-migrate/integrate-api-platform.html).
+
+{% endinfo_block %}
+
+### 404 Not Found for API endpoints
+
+**Symptom:** API requests return 404.
+
+**Possible causes:**
+
+1. Router not configured
+2. Routes not loaded
+3. Wrong URL format
+
+**Solution:**
+
+1. Verify `SymfonyFrameworkRouterPlugin` is registered:
+
+   ```php
+   // RouterDependencyProvider
+   protected function getRouterPlugins(): array
+   {
+       return [
+           new GlueRouterPlugin(),
+           new SymfonyFrameworkRouterPlugin(), // Must be present
+       ];
+   }
+   ```
+
+2. Check API documentation for correct URLs:
+
+   ```bash
+   Storefront: https://glue.mysprykershop.com/
+   Backoffice: https://backoffice.mysprykershop.com/
+   ```
+
+   The interactive API documentation is available at the root URL of each application.
+
+3. Use correct URL format:
+
+   ```bash
+   ❌ /api/v1/customers
```

Comment:
what if I still want to have API with this URL?

Reply:
You can have it, but it is not recommended. You can define url_prefix in the api_platform.php config files.

Probably there are other ways as well. Recommendation is to use the evolution strategy instead.